pkgs/servers/irc/inspircd/default.nix

```diff
@@ -0,0 +1,192 @@
+let
+  # instead of inventing a license compatibility
+  # predicate, let's just whitelist licenses known
+  # to be compatible which are used by actual libcs
+  # in nixpkgs
+  compatible = lib: drv:
+    lib.any (lic: lic == drv.meta.license) [
+      lib.licenses.mit        # musl
+      lib.licenses.lgpl2Plus  # glibc
+    ];
+
+  # compatible if libc is compatible
+  libcModules = [
+    "regex_posix"
+    "sslrehashsignal"
+  ];
+
+  # compatible if libc++ is compatible
+  # TODO(sternenseemann):
+  # we can enable "regex_stdlib" automatically, but only if
+  # we are using libcxxStdenv which is compatible with GPLv2,
+  # since the gcc libc++ uses GPLv3
+  libcxxModules = [
+    "regex_stdlib"
+  ];
+
+  compatibleModules = lib: stdenv: [
+    # GPLv2 compatible dependencies
+    "argon2"
+    "ldap"
+    "mysql"
+    "pgsql"
+    "regex_pcre"
+    "regex_re2"
+    "regex_tre"
+    "sqlite3"
+    "ssl_gnutls"
+  ] ++ lib.optionals (compatible lib stdenv.cc.libc) libcModules;
+
+in
+
+{ lib
+, stdenv
+, fetchFromGitHub
```

Comment:
I’d move the package header to the top, to not confuse any automatic scripts people might run.

Reply:
update, I noticed that it references the modules from above

Reply:
Yep, I also don't like it too much, but it's better than inlining a huge if else thingy